buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlTestPlugin.java

```diff
@@ -0,0 +1,126 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test.rest;
+
+import org.elasticsearch.gradle.plugin.PluginPropertiesExtension;
+import org.elasticsearch.gradle.test.RestIntegTestTask;
+import org.elasticsearch.gradle.testclusters.RestTestRunnerTask;
+import org.elasticsearch.gradle.testclusters.TestClustersPlugin;
+import org.elasticsearch.gradle.util.GradleUtils;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.JavaBasePlugin;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.bundling.Zip;
+
+/**
+ * Apply this plugin to run the YAML based REST tests. This will adda
+ */
+public class YamlTestPlugin implements Plugin<Project> {
+
+    public static final String SOURCE_SET_NAME = "yamlTest";
+
+    @Override
+    public void apply(Project project) {
+
+        if (project.getPluginManager().hasPlugin("elasticsearch.build") == false
+            && project.getPluginManager().hasPlugin("elasticsearch.standalone-rest-test") == false) {
+            throw new InvalidUserDataException(
+                "elasticsearch.build or elasticsearch.standalone-rest-test plugin " + "must be applied before the YAML test plugin"
+            );
+        }
+
+        // to spin up the external cluster
+        project.getPluginManager().apply(TestClustersPlugin.class);
+        // to copy around the yaml tests and json spec
+        project.getPluginManager().apply(RestResourcesPlugin.class);
+
+        // note - source sets are not created via org.elasticsearch.gradle.util.GradleUtils.addTestSourceSet since unlike normal tests
+        // we only want the yamlTestSourceSet on the classpath by default. The yaml tests should be pure black box testing over HTTP and
+        // such it should not need the main class on the class path. Also, there are some special setup steps unique to YAML tests.
+
+        // create source set
+        SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+        SourceSet yamlTestSourceSet = sourceSets.create(SOURCE_SET_NAME);
+
+        // create task - note can not use .register due to the work in RestIntegTestTask's constructor :(
+        RestIntegTestTask yamlTestTask = project.getTasks()
+            .create(
+                SOURCE_SET_NAME,
+                RestIntegTestTask.class,
+                task -> { task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask")); }
+            );
+        yamlTestTask.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+        yamlTestTask.setDescription("Runs the YAML based REST tests against an external cluster");
+
+        // setup task dependency
+        project.getDependencies().add(SOURCE_SET_NAME + "Compile", project.project(":test:framework"));
```

Comment:
For now we need to check if `BuildParams.isInternal` here and use an artifact dependency if not.

Reply:
done